modules/onboarding/src/main/kotlin/edu/stanford/spezi/module/onboarding/OnboardingNavigationEvent.kt

```diff
@@ -8,4 +8,5 @@ sealed class OnboardingNavigationEvent : NavigationEvent {
     data object OnboardingScreen : OnboardingNavigationEvent()
     data object SequentialOnboardingScreen : OnboardingNavigationEvent()
     data object ConsentScreen : OnboardingNavigationEvent()
+    data object ClearBackStackOnboarding : OnboardingNavigationEvent()
```

Comment:
I would propose to reuse the existing event and parametrise it:

```kotlin
data class OnboardingScreen(val clearBackStack: Boolean) : OnboardingNavigationEvent()
```

And you can emit this event with flag true on sign up. You can adapt listening the event in the activity then as follows:

```kotlin
is OnboardingNavigationEvent.OnboardingScreen -> navHostController.navigate(
    Routes.OnboardingScreen,
) {
    if (event.clearBackStack) {
        popUpTo(navHostController.graph.startDestinationId) { inclusive = true }
        launchSingleTop = true // we need this config as well
    }
}
```

Reply:
oh yes that's right, that makes a lot more sense, I don't know why I didn't think of it